pkg/proxy/roundrobin.go

```diff
@@ -210,14 +236,14 @@ func (lb *LoadBalancerRR) OnUpdate(endpoints []api.Endpoints) {
 	for _, endpoint := range endpoints {
 		existingEndpoints, exists := lb.endpointsMap[endpoint.Name]
 		validEndpoints := filterValidEndpoints(endpoint.Endpoints)
-		if !exists || !reflect.DeepEqual(existingEndpoints, validEndpoints) {
+		if !exists || !endpointsAreEqual(existingEndpoints, validEndpoints) {
```

Comment:
To be clear, are you suggesting I add SortSlice and CopySlice to github.com/GoogleCloudPlatform/kubernetes/pkg/util?

I get your point, that you want to copy the unnecessary copy of `validEndpoints`. I also looked a bit into `util.StringSet`. I could call `HasAll` and check the lengths.